packages/block-library/src/term-description/use-term-description.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * WordPress dependencies
+ */
+import { store as coreStore, useEntityProp } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Hook to fetch term description based on context or fallback to template parsing.
+ *
+ * This hook prioritizes context-provided termId and taxonomy, but falls back to
+ * template-based detection when no context is available.
+ *
+ * @param {string|number} termId   The term ID from context
+ * @param {string}        taxonomy The taxonomy name from context
+ */
+export function useTermDescription( termId, taxonomy ) {
+	const [ description, setDescription, fullDescription ] = useEntityProp(
+		'taxonomy',
+		taxonomy,
+		'description',
+		termId
+	);
+
+	// Fallback approach: Parse template slug when no context is available
+	const templateBasedData = useTemplateBasedTermData();
+
+	const hasContext = Boolean( termId && taxonomy );
+
+	return {
+		hasContext,
+		setDescription,
+		termDescription: hasContext
+			? fullDescription?.rendered || description || ''
+			: templateBasedData.termDescription,
+	};
+}
+
+/**
+ * Fallback hook to fetch term data from template context (backward compatibility).
+ * This maintains the same logic as the original implementation for cases where
+ * no termId/taxonomy context is provided.
+ */
+function useTemplateBasedTermData() {
+	const templateSlug = useSelect( ( select ) => {
+		// Access core/editor by string to avoid @wordpress/editor dependency.
+		// eslint-disable-next-line @wordpress/data-no-store-string-literals
+		const { getCurrentPostId, getCurrentPostType, getCurrentTemplateId } =
+			select( 'core/editor' );
```

Comment:
I do think this is a bit messed up. We don't really avoid a `@wordpress/editor` dependency, do we? So why pretend? Looks like something we should address further down the line.

Reply:
I think we're mainly avoiding `@wordpress/block-library` from being dependent on `@wordpress/editor`, as blocks can be loaded into a non-post block editor. We're doing something similar for `useArchiveLabel()`, so if we're going to address this in a different way, then perhaps we could handle both of these cases at once in a separate PR (and probably a few other similar cases).